### ISSUE TITLE
Select: Sort dropdown results by relevance

### ISSUE THREAD

**Issue by vitPinchuk** [grafana/grafana#98186]
**Why is this needed**:

Sort dropdown results by relevance. 

https://developers.grafana.com/ui/latest/index.html?path=/story/forms-select--multi-select-basic

It will sort the matching results so that the most relevant ones.  For example, alphabetically or in reverse order. 
Currently, when searching for options, the options that are returned during the search retain their original order, there is no way to sort the matching results.

Current version: 
![Image](https://github.com/user-attachments/assets/50a76f76-3457-4cb9-b576-4fb288f9b575)

`dev/foo` shows first. According to the options order in Select component

**What would you like to be added**:

New props. for `Select` (`Multiselect`) component. It can be 'string' like `asc` or `desc` or function to implement sorting. 

**Who is this feature for?**

It is for users who want to search options and may posibility to see the result with more relevant order

**Comment by tskarhed** [comment]
I don't think we will introduce a new prop for this, but rather improve the default behavior. Since we have been working on a Select replacement `Combobox`, we might introduce those improvements there instead. I will talk a bit with the team and hear what they have to say.

**Comment by eledobleefe** [comment]
Closing this in favour of [Combobox: Fuzzy search](https://github.com/grafana/grafana/issues/98929)

Hi @vitPinchuk,  
The team agreed to keep the 'Select' component in a 'maintenance mode'. 
Nevertheless, we will take this issue into account when working on the new component's [fuzzy search task](https://github.com/grafana/grafana/issues/98929), which, as you can see, already have 'improving the sorting' in mind.

**Comment by vitPinchuk** [comment]
@eledobleefe @tskarhed Thank you
